viewing_party/party.py

```diff
@@ -1,23 +1,173 @@
+        for j in range(len((user_data)["friends"][i]["watched"])):
+            list_friend_watched.append(user_data["friends"][i]["watched"][j])
+
+    list_user_watched = [item for item in user_data["watched"]]
+
+    list_unique_user_watched = []
+    for item in list_user_watched:
+        if item not in list_friend_watched:
+            list_unique_user_watched.append(item)
+
+    return list_unique_user_watched
+
+def get_friends_unique_watched(user_data):
+    # Method that doesn't use list comprehension, and would work if a friend gave a move a different rating than the user
+    # list_both_watched = []
+    # for i in range(len(user_data["watched"])):
+    #     for j in range(len(user_data["friends"])):
+    #         for k in range(len(user_data["friends"][j]["watched"])):
+    #             if user_data["watched"][i]["title"] == user_data["friends"][j]["watched"][k]["title"] \
+    #                 and user_data["watched"][i]["title"] not in list_both_watched:
+    #                 list_both_watched.append(user_data["watched"][i])
+    
+    # list_friend_watched = []
+    # for j in range(len(user_data["friends"])):
+    #         for k in range(len(user_data["friends"][j]["watched"])):
+    #             list_friend_watched.append(user_data["friends"][j]["watched"][k])
+
+    # list_unique_friend_watched = []
+    # for item in list_friend_watched:
+    #     if item not in list_both_watched and item not in list_unique_friend_watched:
+    #         list_unique_friend_watched.append(item)
+
+    list_friend_watched = []
+    for i in range(len(user_data["friends"])):
+        for j in range(len((user_data)["friends"][i]["watched"])):
+            list_friend_watched.append(user_data["friends"][i]["watched"][j])
+
+    list_user_watched = [item for item in user_data["watched"]]
+
+    list_unique_friend_watched = []
+    for item in list_friend_watched:
+        if item not in list_user_watched and item not in list_unique_friend_watched:
+            list_unique_friend_watched.append(item)
+    
+
+    return list_unique_friend_watched
 # -----------------------------------------
 # ------------- WAVE 4 --------------------
 # -----------------------------------------
 
+def get_available_recs(user_data):
+    friend_recs = get_friends_unique_watched(user_data)
+    recommended_movies = \
+        [item for item in friend_recs if item["host"] in user_data["subscriptions"]]
+    return recommended_movies
+
 # -----------------------------------------
 # ------------- WAVE 5 --------------------
 # -----------------------------------------
 
+def get_new_rec_by_genre(user_data):
+
+    # Create a dictionary of genres and their overall frequency
+    genre_dict = {}
+    for movie_dict in user_data["watched"]:
+        movie_genre = movie_dict["genre"]
+        if movie_genre in genre_dict.keys():
+            genre_dict[movie_genre] += 1
+        else:
+            genre_dict[movie_genre] = 1

```

Comment:
Great use of a frequency dictionary! One small note, the `.keys()` is unneeded here. For a dictionary, `in` checks the keys by default.